src/map.c

```diff
@@ -1312,10 +1640,166 @@ static void map_buttonZoom( unsigned int wid, char* str )
    map_ypos *= map_zoom;
 }
 
+/**
+ * @brief Generates the list of map modes, i.e. commodities that have been seen so far.
+ */
+static void map_genModeList(void){
+   int i,j,k,l;
+   int tot=0;
+   Planet *p;
+   StarSystem *sys;
+   int totGot = 0;
+   if ( commod_known == NULL )
+      commod_known = malloc(sizeof(Commodity*) * commodity_getN());
+   memset(commod_known,0,sizeof(Commodity*)*commodity_getN());
+   for (i=0; i<systems_nstack; i++) {
+      sys = system_getIndex( i );
+      for( j=0 ; j<sys->nplanets; j++){
+	 p=sys->planets[j];
+	 tot+=p->ncommodities;
+	 for( k=0; k<p->ncommodities; k++){
+	    if ( p->commodityPrice[k].cnt > 0 ){/*commodity is known about*/
+	       /* find out which commodity this is */
+	       for ( l=0 ; l<totGot; l++){
+		  if ( p->commodities[k] == commod_known[l] )
+		     break;
+	       }
+	       if ( l == totGot ){
+		  commod_known[totGot] = p->commodities[k];
+		  totGot++;
+	       }
+	       
+	    }
+	 }
+      }
+   }
+   if ( map_modes != NULL ){
+      for ( i=0 ; i<nmap_modes ; i++)
+	 free( map_modes[i] );
+      free ( map_modes );
+   }
+   nmap_modes = 2*totGot + 1;
+   map_modes = calloc( sizeof(char*), nmap_modes );
+   map_modes[0] = strdup("Travel (Default)");
+   
+   for ( i=0; i<totGot; i++ ){
+      l = strlen(commod_known[i]->name) + 7;
+      map_modes[ 2*i + 1 ] = malloc(l);
+      nsnprintf(map_modes[2*i+1],l, "%s: Cost", commod_known[i]->name );
+      l+=6;
+      map_modes[ 2*i + 2 ] = malloc(l);
+      nsnprintf(map_modes[2*i+2],l, "%s: Trade", commod_known[i]->name );
+   }
+}
+
+/**
+ * @brief Updates the map mode list.  This is called when the map update list is clicked.
+ *    Unfortunately, also called when scrolled.
+ *    @param wid Window of the map window.
+ *    @param str Unused.
+ */
+static void map_mode_update( unsigned int wid, char* str )
```

Comment:
Function name should be "map_modeUpdate" to be consistent with the rest of function names.